### PR TITLE
Set tab titles to match legacy tab titles

### LINKED ISSUE #8734: Update browser tab titles
**Describe the bug**
In North Star, the title of the browser tab is always "Title".

**To Reproduce**

1. Enable North Star
2. Navigate around the app
3. Check the browser tab title

**Expected behavior**
The title should be customized based on the civic entity, program, and/or page.

**Screenshots**
Screenshot 2024-09-26 at 3 31 40 PM

### PR DESCRIPTION
**Pull request by ssandbekkhaug**
### Description

Before: tab titles were "Title", in English, regardless of selected language

After: tab titles are the same as legacy pages (e.g. home page is "Find Programs", localized to the user's selected language)

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).

### Issue(s) this completes

Fixes #8734

### CODE REVIEW

**Comment by ssandbekkhaug** [comment]
* **#9026** <a href="https://app.graphite.dev/github/pr/civiform/civiform/9026?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ssandbekkhaug and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: server/app/views/applicant/NorthStarApplicantProgramSummaryView.java [hunk 1]
```diff
@@ -51,6 +53,15 @@ public String render(Request request, Params params) {
             Optional.of(params.profile()),
             params.applicantPersonalInfo(),
             params.messages());
+
+    // Create a string such as "Program appplication summary - Pet Assistance Program"
+    String summarySubstring =
+        params.programType().equals(ProgramType.COMMON_INTAKE_FORM)
+            ? params.messages().at(MessageKey.TITLE_COMMON_INTAKE_SUMMARY.getKeyName())
+            : params.messages().at(MessageKey.TITLE_PROGRAM_SUMMARY.getKeyName());
+    String pageTitle = String.format("%s — %s", summarySubstring, params.programTitle());
```
Comment (swatkat1) [inline server/app/views/applicant/NorthStarApplicantProgramSummaryView.java:62]: qq [non-blocking]- is the programTitle, the localized program name. We have had issue in the past that the admin name gets displayed.

File: server/app/controllers/applicant/ApplicantProgramReviewController.java [hunk 2]
```diff
@@ -209,6 +209,7 @@ public CompletionStage<Result> reviewWithApplicantId(
                         .setSuccessBannerMessage(flashSuccessBannerMessage)
                         .setEligibilityAlertSettings(eligibilityAlertSettings)
                         .setSummaryData(summaryData)
+                        .setProgramType(roApplicantProgramService.getProgramType())
```
Comment (swatkat1) [inline server/app/controllers/applicant/ApplicantProgramReviewController.java:212]: [blocking]-Should add browser tests to it?

Comment (ssandbekkhaug) [reply server/app/controllers/applicant/ApplicantProgramReviewController.java:212]: You're right. Added tests

File: browser-test/src/applicant/navigation/application_navigation_flow.test.ts [hunk 3]
```diff
@@ -92,13 +92,24 @@ test.describe('Applicant navigation flow', () => {
...
-      'review page with North Star enabled',
+      'flows with North Star enabled',
       {tag: ['@northstar']},
       () => {
         test.beforeEach(async ({page}) => {
           await enableFeatureFlag(page, 'north_star_applicant_ui')
         })
 
+        test('validate block edit page title', async ({
+          page,
```
Comment (swatkat1) [inline browser-test/src/applicant/navigation/application_navigation_flow.test.ts:103]: [non- blocking]- but it's good idea to move this a function in support/* folder and pass in the title as an input parameter. This function will have good reusability in the future.

**Review by swatkat1 — APPROVED** [review]
Thank you for adding tests!

File: browser-test/src/applicant/navigation/application_navigation_flow.test.ts (hunk 3 above)
Comment (ssandbekkhaug) [reply browser-test/src/applicant/navigation/application_navigation_flow.test.ts:103]: Done

**Review by swatkat1 — APPROVED** [review]

**Pull request merged**